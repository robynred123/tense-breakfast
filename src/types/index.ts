export interface ReducerState {
  therapists: TherapistInfo[];
  filteredTherapists: TherapistInfo[];
  filterOptions: FilterOptions;
  mobileFilter: boolean;
  error: string | null;
}

export interface TherapistInfo {
  id: string;
  firstName: string;
  lastName: string;
  appointment_types: AppointmentType[];
  appointment_mediums: AppointmentMedium[];
  specialisms: Specialisms[];
}

export interface DateRange {
  start: string | null;
  end: string | null;
}

export interface AvailabilityData {
  id: string;
  datetime: string;
}

export interface Availabilities {
  [therapistId: string]: AvailabilityData[];
}

export type Type = 'date' | 'type' | 'specialism';

export type AppointmentType = 'one_off' | 'consultation';
export type AppointmentMedium = 'phone' | 'video';
export type Specialisms = string;

export type FilterOptions = {
  appointmentType: AppointmentType[];
  dateRange: DateRange;
  specialisms: Specialisms[];
};

export type BookingOptions = {
  therapistId: string;
  appointmentType: AppointmentType | null;
  appointmentMedium: AppointmentMedium | null;
  appointmentDate: string;
};
