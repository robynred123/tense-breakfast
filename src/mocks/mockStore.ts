import configureStore from 'redux-mock-store';
import { Store, AnyAction } from 'redux';
import { FilterOptions, ReducerState } from '../types/index';

export const filterOptions: FilterOptions = {
  appointmentType: [],
  dateRange: {
    start: null,
    end: null,
  },
  specialisms: [],
};
export const initialState: ReducerState = {
  therapists: [],
  filteredTherapists: [],
  filterOptions,
  mobileFilter: false,
  error: null,
};

export const mockStore = (initialState: ReducerState) => {
  const mockStore = configureStore();
  let store: Store<any, AnyAction>;
  store = mockStore({ one: initialState });
  return store;
};
