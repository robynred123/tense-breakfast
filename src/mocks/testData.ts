import { TherapistInfo } from '../types';

export const therapistOne: TherapistInfo = {
  id: '063eb398-45f7-4a9c-a5d3-e95d14c05c44',
  firstName: 'Albert',
  lastName: 'Einstein',
  appointment_types: ['one_off'],
  appointment_mediums: ['phone'],
  specialisms: ['Alcoholism', 'Gender identity', 'Gambling Addiction'],
};

export const therapistTwo: TherapistInfo = {
  id: '154aa58b-4c25-4d64-a7a2-9288a7d94468',
  firstName: 'Jackie',
  lastName: 'Chan',
  appointment_types: ['consultation', 'one_off'],
  appointment_mediums: ['video'],
  specialisms: ['Gender identity', 'Alcoholism'],
};
