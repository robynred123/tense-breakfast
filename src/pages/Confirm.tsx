import { Box, Grid, Typography } from '@mui/material';
import moment from 'moment';
import { useNavigate, useLocation } from 'react-router-dom';
import { bookingRequest } from '../actions/actions';
import { ButtonComponent } from '../components/Button';
import { useAppDispatch } from '../store';
import { AvailabilityData, BookingOptions } from '../types';

interface State {
  bookingOptions: BookingOptions;
  therapistName: string;
  dateTime: AvailabilityData;
}

export const ConfirmPage = () => {
  const { state } = useLocation();
  const { bookingOptions, therapistName, dateTime } = state as State;
  const navigate = useNavigate();
  const dispatch = useAppDispatch();

  return (
    <Box sx={{ flexGrow: 1, width: 1, flexDirection: 'row', wrap: 'nowrap' }}>
      <Grid
        item
        md={12}
        lg={12}
        sm={12}
        sx={{
          justifyContent: 'center',
          textAlign: 'center',
        }}
      >
        <Typography variant='h2' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
          Confirm Appointment
        </Typography>

        <Grid item sx={{ flexDirection: 'row', display: 'flex', justifyContent: 'center' }}>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            Therapist:
          </Typography>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            {therapistName}
          </Typography>
        </Grid>

        <Grid item sx={{ flexDirection: 'row', display: 'flex', justifyContent: 'center' }}>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            Date/Time:
          </Typography>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            {moment(dateTime.datetime).format('LLLL')}
          </Typography>
        </Grid>

        <Grid item sx={{ flexDirection: 'row', display: 'flex', justifyContent: 'center' }}>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            Type:
          </Typography>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            {bookingOptions.appointmentType}
          </Typography>
        </Grid>

        <Grid item sx={{ flexDirection: 'row', display: 'flex', justifyContent: 'center' }}>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            Medium:
          </Typography>
          <Typography variant='h5' fontFamily={'lato, sans-serif'} sx={{ padding: '1em' }}>
            {bookingOptions.apppointmentMedium}
          </Typography>
        </Grid>
      </Grid>
      <Grid
        container
        sx={{ justifyContent: 'space-evenly', alignContent: 'center', padding: '2em' }}
      >
        <ButtonComponent
          onClick={() => navigate(-1)}
          text='Cancel'
          buttonColour={'grey'}
          disabled={false}
          width={'25%'}
        />
        <ButtonComponent
          onClick={() => dispatch(bookingRequest(bookingOptions, navigate))}
          text='OK'
          buttonColour={'gradient'}
          disabled={false}
          width={'25%'}
        />
      </Grid>
    </Box>
  );
};
