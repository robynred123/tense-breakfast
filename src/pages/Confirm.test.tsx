/* eslint-disable testing-library/await-async-query */
import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { mockStore, initialState } from '../mocks/mockStore';
import { ConfirmPage } from './Confirm';
import { Router } from 'react-router';
import { createBrowserHistory } from 'history';
import { therapistOne } from '../mocks/testData';
import { BookingOptions } from '../types';

describe('<ConfirmPage />', () => {
  const mockBookingoptions: BookingOptions = {
    therapistId: therapistOne.id,
    appointmentType: 'one_off',
    appointmentMedium: 'phone',
    appointmentDate: 'fe69895c-8a90-426b-89d1-de9043e3bb99',
  };

  const therapistName: string = `${therapistOne.firstName} ${therapistOne.lastName}`;

  const mockUseLocationValue = {
    pathname: '/Confirm',
    search: '',
    hash: '',
    state: {
      bookingOptions: mockBookingoptions,
      therapistName: therapistName,
      dateTime: {
        id: 'fe69895c-8a90-426b-89d1-de9043e3bb99',
        datetime: '2021-08-17T07:00:00.000Z',
      },
    },
  };

  const history = createBrowserHistory();

  beforeEach(() => {
    jest.resetAllMocks();
  });

  it('Should render the confirm Page successfully with expected data', () => {
    const store = mockStore(initialState);
    const view = renderer.create(
      <Provider store={store}>
        <Router navigator={history} location={mockUseLocationValue}>
          <ConfirmPage />
        </Router>
      </Provider>
    );
    expect(view.toJSON()).toMatchSnapshot();
    expect(view.root.findByType('h2').props.children).toEqual('Confirm Appointment');
    expect(view.root.findAllByType('h5')[0].props.children).toEqual('Therapist:');
    expect(view.root.findAllByType('h5')[1].props.children).toEqual(therapistName);
    expect(view.root.findAllByType('h5')[2].props.children).toEqual('Date/Time:');
    expect(view.root.findAllByType('h5')[3].props.children).toEqual(
      'Tuesday, August 17, 2021 8:00 AM'
    );
    expect(view.root.findAllByType('h5')[4].props.children).toEqual('Type:');
    expect(view.root.findAllByType('h5')[5].props.children).toEqual('one_off');
    expect(view.root.findAllByType('h5')[6].props.children).toEqual('Medium:');
    expect(view.root.findAllByType('h5')[7].props.children).toEqual('phone');

    expect(view.root.findAllByType('button')[0].props.children).toEqual([
      [undefined, 'Cancel', undefined],
      null,
    ]);
    expect(view.root.findAllByType('button')[1].props.children).toEqual([
      [undefined, 'OK', undefined],
      null,
    ]);
  });
});
