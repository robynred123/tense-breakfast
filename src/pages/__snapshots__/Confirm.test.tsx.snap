// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ConfirmPage /> Should render the confirm Page successfully with expected data 1`] = `
<div
  className="MuiBox-root css-1rq3dmv"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12 MuiGrid-grid-md-12 MuiGrid-grid-lg-12 css-1aiqtfq-MuiGrid-root"
  >
    <h2
      className="MuiTypography-root MuiTypography-h2 css-s4howv-MuiTypography-root"
    >
      Confirm Appointment
    </h2>
    <div
      className="MuiGrid-root MuiGrid-item css-170b5gt-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Therapist:
      </h5>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Albert Einstein
      </h5>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-170b5gt-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Date/Time:
      </h5>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Tuesday, August 17, 2021 8:00 AM
      </h5>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-170b5gt-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Type:
      </h5>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        one_off
      </h5>
    </div>
    <div
      className="MuiGrid-root MuiGrid-item css-170b5gt-MuiGrid-root"
    >
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        Medium:
      </h5>
      <h5
        className="MuiTypography-root MuiTypography-h5 css-nivqii-MuiTypography-root"
      >
        phone
      </h5>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-container css-gqqfs-MuiGrid-root"
  >
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-1mzfrza-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      Cancel
    </button>
    <button
      className="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-ndj2fk-MuiButtonBase-root-MuiButton-root"
      disabled={false}
      onBlur={[Function]}
      onClick={[Function]}
      onContextMenu={[Function]}
      onDragLeave={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseDown={[Function]}
      onMouseLeave={[Function]}
      onMouseUp={[Function]}
      onTouchEnd={[Function]}
      onTouchMove={[Function]}
      onTouchStart={[Function]}
      tabIndex={0}
      type="button"
    >
      OK
    </button>
  </div>
</div>
`;
